sale_promotion_discount_in_field/models/sale_order.py

```diff
@@ -0,0 +1,133 @@
+# Copyright 2022 Ooops404
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    def _create_new_no_code_promo_reward_lines(self):
+        """
+        Override core method to save program with new discount apply rule
+        """
+        self.ensure_one()
+        order = self
+        programs = order._get_applicable_no_code_promo_program()
+        programs = (
+            programs._keep_only_most_interesting_auto_applied_global_discount_program()
+        )
+        if any([program.reward_type == "discount_line" for program in programs]):
```

Comment:
My feel is this isn't right. What happens with product and delivery rewards? Why isn't `_get_reward_values_discount` overrinding enough? There you'll have the program and you'll be able to prepare the line values accordingly.

Reply:
we cannot use the `_get_reward_values_discount` because in original method `_create_new_no_code_promo_reward_lines` expects some values to save to order_line:

`self.write({'order_line': [(0, False, value) for value in self._get_reward_line_values(program)]})`

I will try to another method

Reply:
I agree with you, I changed the method